pilot/pkg/networking/plugin/authn/util.go

```diff
@@ -0,0 +1,28 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package authn
+
+import (
+	meshconfig "istio.io/api/mesh/v1alpha1"
+	"istio.io/istio/pilot/pkg/features"
+)
+
+func trustDomainsForValidation(meshConfig *meshconfig.MeshConfig) []string {
+	var trustDomains []string
+	if !features.SkipValidateTrustDomain.Get() {
+		trustDomains = append(meshConfig.TrustDomainAliases, meshConfig.TrustDomain)
```

Comment:
Can you make a copy of the `meshConfig.TrustDomainAliases`? Otherwise it's possible the same underline slice is shared and a change (should not happen but could) to the trustDomains could change the value in the meshconfig.

Reply:
> Also realized one more thing, can you set SkipValidateTrustDomain to always true to disable the old trust domain validation flow in the istio_authn filter:

Is this really necessary? features.SkipValidateTrustDomain defaults to true unless user sets the env var? If I remove this wouldn't that remove the ability to disable Trust Domain Validation?

Reply:
it's needed (not optional) because the trust_domain_alias is not checked in the istio_authn filter which means it will still be rejected by the filter even it is allowed in the transport socket level.

We should only have 1 implementation of trust domain validation, with the transport socket support from this PR, we should disable the old implementation in the istio_authn filter.

Reply:
Ah gotchu